src/analyze/analyze.c

```diff
@@ -85,6 +100,10 @@ static unsigned arg_iterations = 1;
 
 STATIC_DESTRUCTOR_REGISTER(arg_dot_from_patterns, strv_freep);
 STATIC_DESTRUCTOR_REGISTER(arg_dot_to_patterns, strv_freep);
+STATIC_DESTRUCTOR_REGISTER(arg_dot_dependency_colors_styles, strv_freep);
+
+static char *arg_dependency_color[_UNIT_DEPENDENCY_MAX] = {};
+static char *arg_dependency_style[_UNIT_DEPENDENCY_MAX] = {};
```

Comment:
using STATIC_DESTRUCTOR_REGISTER(), but not sure if freep is correct here

Reply:
given they are static arrays with malloc'ed elements, you need to define a new function, eg dependency_destroy, and use it instead of freep. The function argument will be char ***